src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,133 @@
+    pipeline_ = create_pipeline(config_str);
+    config_json_ = nlohmann::json::parse(config_str);
+
+    // extract resolution of the frames and create cv::Mat placeholders
+    set_resolution();
+    create_frame_holders();
+}
+
+void DepthAI::set_resolution()
+{
+    // if camera is defined check if resolution of stereo or rgb is defined. else set to default
+    if (config_json_.contains("camera")) {
+        auto& camera_conf_obj = config_json_.at("camera");
+        /*
+         * if rgb res is set in config_json_ fetch it to set 
+         * the height of the image else set the default
+         */
+        if (camera_conf_obj.contains("rgb")) {
+            auto& rgb_camera_conf_obj = camera_conf_obj.at("rgb");
+            rgb_height_ = rgb_camera_conf_obj.at("resolution_h").get<int32_t>();
+            // _rgb_width = width_to_height_map_[_rgb_height];
+        } else {
+            rgb_height_ = 3040;
+            rgb_width_ = 4056;
+        }
+        /*
+         * if stereo camera res is set in config_json_ fetch it to set 
+         * the height of the image else set the default
+         */
+        if (camera_conf_obj.contains("mono")) {
+            auto& mono_camera_conf_obj = camera_conf_obj.at("mono");
+            mono_height_ = mono_camera_conf_obj.at("resolution_h").get<int32_t>();
+            auto it = width_to_height_map_.find(mono_height_);
+            mono_width_ = it->second;
+        } else {
+            mono_height_ = 720;
+            mono_width_ = 1280;
+        }
+    } else {
+        rgb_height_ = 3040;
+        rgb_width_ = 4056;
+        mono_height_ = 720;
+        mono_width_ = 1280;
+    }
+}
+
+void DepthAI::create_frame_holders()
+{
+    // creating place holders for all the streams enabled in config
+    if (config_json_.contains("streams")) {
+        for (const auto& it : config_json_.at("streams")) {
+            if (it.is_string()) {
+                if (it.get<std::string>() == "metaout")
+                    continue;
+                if (it.get<std::string>() == "previewout") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_stream_holder["previewout"] = img;
+                }
+            } else {
+                const auto& name = it.at("name").get<std::string>();
+                if (name == "previewout") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_stream_holder["previewout"] = img;
+                } else if (name == "depth") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC1);
+                    image_stream_holder["depth"] = img;
+                } else if (name == "disparity_color") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC3);
+                    image_stream_holder["disparity_color"] = img;
+                }
```

Comment:
This entire function doesn't follow the style of the rest of this file. Can you put `{` in their own lines?

Reply:
the whole file follows the sane line except the function's `{`. I will change that instead of all the braces